internal/resources/otelcollector/core.go

```diff
@@ -129,7 +129,7 @@ func makeMetricsService(name types.NamespacedName) *corev1.Service {
 	}
 }
 
-func makeDenyPprofNetworkPolicy(name types.NamespacedName) *networkingv1.NetworkPolicy {
+func makeDenyPprofNetworkPolicy(name types.NamespacedName, allowedPorts []intstr.IntOrString) *networkingv1.NetworkPolicy {
```

Comment:
Let's make the function name more generic (even though, the policy still has pprof in the name).

Reply:
done